### PR TITLE
Update to 1.8.1

### PR DESCRIPTION
**Pull request by martin-ejdestig**
I assume this should be put on a 1.8.1 branch?

### CODE REVIEW

**Comment by martin-ejdestig** [comment]
@jpakkane @sarum9in How can I test that what will end up on https://wrapdb.mesonbuild.com/gtest will be correct?

**Comment by sarum9in** [comment]
Please rebase it on the branch I pointed it to, which is exact copy of `master`. Something is wrong with diffs at the moment.

> How can I test

I usually just use `mesonwrap review` tool from https://github.com/mesonbuild/wrapweb

**Comment by martin-ejdestig** [comment]
> Please rebase it on the branch I pointed it to, which is exact copy of `master`. Something is wrong with diffs at the moment.

I do not think it looks incorrect. The commits listed now is what `git log master..origin/1.8.0` shows as well for me locally. The master branch does not have any of those commits.

Did you perhaps mean to base the 1.8.1 branch on the 1.8.0 branch? Would make sense, I think. Otherwise I will have to copy the files from the 1.8.0 branch or take all commits after master and 1.8.0 diverged.

Will wait with doing anything until I know what you want me to do. :)

> I usually just use `mesonwrap review` tool from https://github.com/mesonbuild/wrapweb

I ran `./mesonwrap.py review gtest --pull_request=9` and it looks OK to me. Thanks.

File: readme.txt [hunk 1]
```diff
@@ -1,5 +1,3 @@
 This repository contains a Meson build definition for project gtest.
 
-All files in this repository have the same license as the original project.
```
Comment (sarum9in) [inline readme.txt]: This what looks surprising to me... Why do you change this file?

Comment (sarum9in) [reply readme.txt]: https://github.com/mesonbuild/gtest/blob/1.8.1/readme.txt does not contain "- All files..." phrase at all.

Comment (martin-ejdestig) [reply readme.txt]: Hum, did you read what I wrote about the 1.8.1 branch being based on master and so it does not have any of the commits on the 1.8.0 branch? (Well, except for the "Created repository for project gtest." one.)

This particular line change was done in the current tip of the 1.8.0 branch:

https://github.com/mesonbuild/gtest/commit/38ab9376eadce060bc18012d5555e2266fe764e6

So, not changed by me.

So, should the 1.8.1 branch not be based on 1.8.0 instead of master? If this is not done, either all commits (before my last one) will have to be cherry picked to 1.8.1 or I will have copy all files from the 1.8.0 branch and do the bump.

I think the best option would be to base the 1.8.1 branch on the 1.8.0 branch to preserve history.

Comment (martin-ejdestig) [reply readme.txt]: Or what am I missing? Perhaps I misunderstand what you mean...

Comment (martin-ejdestig) [reply readme.txt]: ping @sarum9in ?

Can you just repoint the `1.8.1` branch to the `1.8.0` branch instead of `master`? Then I will rebase this pr so only my changes are visible and history will be preserved.

Or will changing `1.8.1` mess things up for https://wrapdb.mesonbuild.com/ since it has already been created and "scraped". (Or something, I have no idea how the page is actually generated. :)

Comment (sarum9in) [reply readme.txt]: > Can you just repoint the 1.8.1 branch to the 1.8.0 branch

No, because if I do that wrapdb will not create a new version 1.8.1.

The thing I am referring to, even though the code you are requesting to merge is pointing to 1.8.1, it tries to delete some lines from `readme.txt` that do not exist. See https://github.com/mesonbuild/gtest/blob/1.8.1/readme.txt and the line I commented. Either I misunderstand how github works or there is something wrong with the merge itself.

I am not going to preserve the history anyway as the general consensus around wraps that we want to keep revisions from 1 for a new version, and current implementation counts from the first commit with `upstream.wrap`. I proposed to use [[wrap version]](https://github.com/mesonbuild/wrapweb/blob/master/mesonwrap/gitutils.py#L45) tag to mark a new version to reset revision counting but this didn't become popular, so the usual approach is to just squash everything.

So TL;DR I am going to debug the git history this weekend to figure out why there is an unexpected diff I see on github.

Comment (martin-ejdestig) [reply readme.txt]: > > Can you just repoint the 1.8.1 branch to the 1.8.0 branch
> 
> No, because if I do that wrapdb will not create a new version 1.8.1.
> 
> The thing I am referring to, even though the code you are requesting to merge is pointing to 1.8.1, it tries to delete some lines from `readme.txt` that do not exist. See https://github.com/mesonbuild/gtest/blob/1.8.1/readme.txt and the line I commented. Either I misunderstand how github works or there is something wrong with the merge itself.

Yes, I think it looked correct. The branch I posted was branched from 1.8.0, not master. Nothing was wrong with GitHub. Easy to see if you looked at the branches with e.g. `git log --graph --oneline --all`, `gitk --all` or some other git history visualization tool.

> I am not going to preserve the history anyway as the general consensus around wraps that we want to keep revisions from 1 for a new version, and current implementation counts from the first commit with `upstream.wrap`. I proposed to use [[wrap version]](https://github.com/mesonbuild/wrapweb/blob/master/mesonwrap/gitutils.py#L45) tag to mark a new version to reset revision counting but this didn't become popular, so the usual approach is to just squash everything.

OK then. I rebased the branch on to master/1.8.1, squashed everything and force pushed.

> So TL;DR I am going to debug the git history this weekend to figure out why there is an unexpected diff I see on github.

It was not unexpected. (See above. ;) But now you do not have to investigate since I force pushed it away. :)

**Comment by martin-ejdestig** [comment]
Thanks for merging. :)

>> Can you just repoint the 1.8.1 branch to the 1.8.0 branch
> No, because if I do that wrapdb will not create a new version 1.8.1.

By the way, this is something that might be worth looking into for the wrapdb webpage... ? Would make a lot of sense, I think, to be able to branch of new version branch from an older one instead of copying files and losing history.
